src/client.ts

```diff
@@ -3940,76 +3940,33 @@ export class MatrixClient extends EventEmitter {
      * @return {Promise} Resolves: Object (currently empty)
      * @return {module:http-api.MatrixError} Rejects: with an error response.
      */
-    public unban(roomId: string, userId: string, callback?: Callback): Promise<void> {
+    public unban(roomId: string, userId: string, callback?: Callback): Promise<{}> {
         // unbanning != set their state to leave: this used to be
         // the case, but was then changed so that leaving was always
         // a revoking of privilege, otherwise two people racing to
-        // kick / ban someone could end up banning and then un-banning
-        // them.
-        const path = utils.encodeUri("/rooms/$roomId/unban", {
-            $roomId: roomId,
-        });
-        const data = {
-            user_id: userId,
-        };
-        return this.http.authedRequest(
-            callback, "POST", path, undefined, data,
-        );
+        // kick / ban someone could end up banning and then un-banning them.
+        return this.membershipChange(roomId, userId, "unban", undefined, callback);
```

Comment:
unban is not a membership: https://github.com/vector-im/element-web/issues/17510

Reply:
Okay for `forget` but for ban/unban/kick/invite a re-usable helper seems clean, they all have the same signature (some omit reason) and instead of membership it could be `verb`